### PR TITLE
Bump emitter after support compiler v0.44

### PR DESCRIPTION
**Pull request by MaryGao**
- Bump the emitter version
- Switch to pnpm workspace
- Switch to peer dependency in emitter

### CODE REVIEW

File: packages/autorest.typescript/package.json [hunk 1]
```diff
@@ -74,7 +74,7 @@
     "source-map-support": "^0.5.16",
     "ts-morph": "^15.1.0",
     "@azure/core-auth": "^1.3.2",
-    "@azure-tools/rlc-common": "^0.12.0"
+    "@azure-tools/rlc-common": "workspace:^0.13.0"
```
Comment (qiaozha) [inline packages/autorest.typescript/package.json:77]: why ? Does it still work if we released autorest.typescript and typespec-ts ?

File: packages/typespec-ts/package.json [hunk 2]
```diff
@@ -58,19 +58,20 @@
     "pacote": "15.1.3",
     "@types/pacote": "11.1.5"
   },
+  "peerDependencies": {
+    "@azure-tools/typespec-azure-core": ">=0.30.0 <1.0.0",
+    "@azure-tools/typespec-client-generator-core": ">=0.30.0 <1.0.0",
+    "@typespec/compiler": ">=0.44.0 <1.0.0",
+    "@typespec/http": ">=0.44.0 <1.0.0",
+    "@typespec/rest": ">=0.44.0 <1.0.0",
+    "@typespec/versioning": ">=0.44.0 <1.0.0"
```
Comment (qiaozha) [inline packages/typespec-ts/package.json:67]: why these libraries are not necessarily required for typescript emiter ? https://docs.npmjs.com/cli/v9/configuring-npm/package-json#peerdependencies  @joheredi Could you help confirm if this is ok with you ?

Comment (xirzec) [reply packages/typespec-ts/package.json:67]: We normally recommend against peer deps for most packages, due to the complexity and confusion surrounding how to satisfy them. I'm not sure if TypeSpec decided differently for the purposes of emitters.

@bterlson can you help offer guidance?

File: .scripts/nightly.yml [hunk 3]
```diff
@@ -1,29 +1,30 @@
+trigger: none
 schedules:
   - cron: "0 0 * * *"
```
Comment (qiaozha) [inline .scripts/nightly.yml:3]: maybe 0 0 is not a very good time to start the task.

Comment (MaryGao) [reply .scripts/nightly.yml:3]: This would be triggered at midnight in a daily schedule.

Comment (MaryGao) [inline .scripts/nightly.yml:3]:
```suggestion
  - cron: "0 16 * * *"
```

File: .scripts/nightly.yml [hunk 4]
```diff
@@ -1,29 +1,30 @@
+trigger: none
 schedules:
-  - cron: "0 0 * * *"
+  - cron: "0 16 * * *"
```
Comment (MaryGao) [inline .scripts/nightly.yml:3]:
```suggestion
  - cron: "0 12 * * *"
```

**Review by qiaozha — APPROVED** [review]
I will approve this for now, @xirzec  let me know if you have other concerns.

**Pull request merged**